### ISSUE TITLE
Download 1.0.0-beta.3 Is Missing Files

### ISSUE THREAD

**Issue by armouredking** [humhub/humhub#1393]
I am not using the git / composer version. Updating per documentation by replacing files and running yii migration at command line.

I can successfully replace files and run command line, however my install no longer works. Logs from the server indicate the following files are missing:

> /var/www/humhub/resources/font-awesome/css/font-awesome.min.css
> /var/www/humhub/js/jquery.timeago.js
> /var/www/humhub/js/jquery.knob.min.js
> /var/www/humhub/js/jquery.nicescroll.min.js
> /var/www/humhub/js/jquery.placeholder.js
> /var/www/humhub/js/jquery.iframe-transport.js
> /var/www/humhub/js/jquery.ui.widget.js
> /var/www/humhub/js/jquery.fileupload.js

The files exist in the 1.0.0-beta.2 download, but do not exist in the -beta.3 download file ( zip or tar ). Assuming oversight for missing files.

**Comment by 0berst** [comment]
As I understand these files are installed at setup and are not longer located in /js.
You will find an /assets folder with generated subfolders in which the missing js files should be located.

If this is not the case something went wrong at setup. Maybe just try it again?

I also did the upgrade with the downloaded archive and not with git/composer and all went great.

**Comment by armouredking** [comment]
How did you _update_ and run setup? That's initial install. That's not part of the _[manual upgrade directions](https://www.humhub.org/docs/guide-admin-updating.html)_ currently included with Humhub. I don't use the automatic updater since it too is in Beta; it's not as if the manual process is complicated. I don't reinstall every time it goes up a version. I just follow the directions, which has worked until now.

I didn't look around in other folders to find the missing files, I just took the ones from beta.2 and placed them in beta.3 where they were missing to get things to progress forward. My /assets folder is filled with a number of randomly generated keystring folders with various contents ( which I think belong to individual spaces / modules, but not 100% sure ). In any case, the missing files are not located in /assets for me, and they should be located in /js during a 'running' instance because that's where the server is calling them from and without them present caused the app to fail to work ( streams wouldn't show and no Font Awesome so the menus were all messed up ).

**Comment by 0berst** [comment]
You are right. I upgraded from 0.11.2 to 1.0.0-beta.3 so i needed to do a [new install](https://www.humhub.org/docs/guide-admin-updating-020.html).

I don't know your setup. All I can say is that in my basicly fresh install the JS files are in /assets/*some random string*/*.js and the page source references them from there. And as I see it this applies exactly to your missing files.

Here is an excerpt from my HumHubs page source code:
```html
...
<link href="/humhub/assets/3933f9b3/css/font-awesome.min.css" rel="stylesheet">
...
<script src="/humhub/assets/43500adf/jquery.nicescroll.min.js"></script>
<script src="/humhub/assets/833b058d/jquery.timeago.js"></script>
<script src="/humhub/assets/833b058d/locales/jquery.timeago.de.js"></script>
<script src="/humhub/assets/1277954/dist/jquery.knob.min.js"></script>
<script src="/humhub/assets/42bf3591/ui/minified/widget.min.js"></script>
<script src="/humhub/assets/74d8ce00/jquery.placeholder.min.js"></script>
<script src="/humhub/assets/a471d06f/js/jquery.fileupload.js"></script>
<script src="/humhub/assets/a471d06f/js/jquery.iframe-transport.js"></script>
...
```
All the other referenced files are in /js, /css or /resources.

Did you get the errors in your webservers or HumHubs logs? Do you use a custom theme?

**Comment by armouredking** [comment]
No custom theme. As stated in the OP, server logs. The app error logs do not report any issues on the day in question ( some logs on the 27th of Dec, and none again until today; seems to be a recurring trend on Sundays about 404 of manifest.json; seems to be a nuisance alert ). Normal install on Debian 8 with NGINX; Self Test / Update and Cron Job settings report no issues and run times as expected. This path started at 0.20, and has progressed up from beta.1 to .2 and finally now .3. Honestly, the app itself has been without major issue till now, it just worked ( minor bugs I ran into like birthdays not withstanding ). This is the first time the app failed in a dramatic fashion.

I'll probably have a go at a reinstall if the next upgrade follows the same suit. The upgrade process as specified in documentation worked for every update until beta.3 without issue, hence this bug report. Nothing has changed in the server environment the entire time.

**Comment by luke-** [comment]
The files should be in folder 'protected/vendor/bower' e.g. jquery-nicescroll.
These will be automatically published to /assets on request.

**Comment by armouredking** [comment]
So I figured this out finally. It's a caching problem. I cleared the NGINX caches but I forgot that Humhub uses it's own file cache. The file cache was using older references not being pointed following the update. They were still referencing the /js folder instead of the /assets folder.

Might need to add a comment to clear file based caching on upgrade or add an auto. The updater module might already have that built in, unsure.
